mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -1217,7 +1221,10 @@ convertOmpParallel(omp::ParallelOp opInst, llvm::IRBuilderBase &builder,
       assert(phis.size() == 1 &&
              "expected one value to be yielded from the "
              "reduction neutral element declaration region");
-
+      if (initBlock)
```

Comment:
Why not use a separate init block for the by-val case as well? I know we currently happen to only have simple allocas in the init region of by val reductions. But this isn't what the by-ref attribute (or not) is guaranteeing.

Reply:
Good point. I wanted to ONLY fix the things that actually were broken - doing this "always" meant fixing another couple of places.